xbmc/utils/EGLUtils.cpp

```diff
@@ -596,3 +590,48 @@ bool CEGLContextUtils::TrySwapBuffers()
 
   return (eglSwapBuffers(m_eglDisplay, m_eglSurface) == EGL_TRUE);
 }
+
+int32_t CEGLContextUtils::GetBufferAge()
+{
+  int32_t bufferAge = 2;
+  if (CEGLUtils::HasExtension(m_eglDisplay, "EGL_KHR_partial_update"))
+    eglQuerySurface(m_eglDisplay, m_eglSurface, EGL_BUFFER_AGE_KHR, &bufferAge);
+  else if (CEGLUtils::HasExtension(m_eglDisplay, "EGL_EXT_buffer_age"))
+    eglQuerySurface(m_eglDisplay, m_eglSurface, EGL_BUFFER_AGE_EXT, &bufferAge);
+
+  return bufferAge;
+}
+
+bool CEGLContextUtils::HasDamagedRegionSupport()
+{
+  return CEGLUtils::HasExtension(m_eglDisplay, "EGL_KHR_partial_update");
```

Comment:
Is it necessary to query the extensions every time or can this be cached? E.g. in a static variable.

Reply:
Currently, this is normal practice all over the place.

That's something I have on my agenda for GL/EGL extensions. I was thinking about doing the string compare at startup, so that we only have to query an enum. But I feel that this is something for another day.